src/services/accession.jl

```diff
@@ -0,0 +1,217 @@
+export Accession, browse
+
+# Accession numbers (S::Symbol database name, T::DataType encoding type)
+immutable Accession{S,T}
+    # encoded accession number
+    data::T
+end
+
+Accession(s::String) = parse(Accession, s)
+
+if VERSION >= v"v0.4-"
+    Base.call{S}(::Type{Accession{S}}, s::String) = parse(Accession{S}, s)
+end
+
+=={S,T}(x::Accession{S,T}, y::Accession{S,T}) = x.data == y.data
+=={S}(x::Accession{S}, y::String) = x == parse(Accession{S}, y)
+=={S}(x::String, y::Accession{S}) = parse(Accession{S}, x) == y
+hash(x::Accession) = hash(x.data)
+
+# smart parser (should not be used outside of an interactive session)
+function parse(::Type{Accession}, s::String)
+    if ismatch(Accession{:RefSeq}, s)
+        return parse(Accession{:RefSeq}, s)
+    elseif ismatch(Accession{:CCDS}, s)
+        return parse(Accession{:CCDS}, s)
+    elseif ismatch(Accession{:Ensembl}, s)
+        return parse(Accession{:Ensembl}, s)
+    elseif ismatch(Accession{:GeneOntology}, s)
+        return parse(Accession{:GeneOntology}, s)
+    end
+    error("cannot guess accession number type")
+end
+
+# parser of decimal integer; this is very common among accession numbers
+function parse_decimal_int(s::String, start::Int=1, stop::Int=endof(s))
```

Comment:
Can you include an explanation of *why* this shouldn't be used outside an interactive session?